src/bindings/python/src/pyopenvino/graph/dict_attribute_visitor.cpp

```diff
@@ -274,6 +274,18 @@ void util::DictAttributeSerializer::on_adapter(const std::string& name, ov::Valu
     if (m_attributes.contains(name)) {
         OPENVINO_THROW("No AttributeVisitor support for accessing attribute named: ", name);
     }
+    
+    if (auto _adapter = dynamic_cast<ov::AttributeAdapter<std::shared_ptr<ov::op::util::Variable>>*>(&adapter)) {
+        m_attributes[name.c_str()] = _adapter->get()->get_info().variable_id;
+    } else if (auto _adapter = dynamic_cast<ov::AttributeAdapter<ov::PartialShape>*>(&adapter)) {
```

Comment:
This is a bit complex situation that i have to explain in detail:

I was once trying to define a specialized `on_adapter()` for class `DictAttributeSerializer` like following (lets take `ov::PartialShape` as an example argument):

```cpp
// src/bindings/python/src/pyopenvino/graph/dict_attribute_visitor.hpp
class DictAttributeSerializer : public ov::AttributeVisitor {
public:
    explicit DictAttributeSerializer(const std::shared_ptr<ov::Node>& node);
    ...
    void on_adapter(const std::string& name, ov::ValueAccessor<ov::PartialShape>& adapter);
    ...
};
```

And i simply implement it with some prints:

```cpp
// src/bindings/python/src/pyopenvino/graph/dict_attribute_visitor.cpp
void util::DictAttributeSerializer::on_adapter(const std::string& name, ov::ValueAccessor<ov::PartialShape>& adapter) {
    std::cout << "on_adapter(ov::AttributeAdapter<ov::PartialShape>)" << std::endl;
}
```

However, when i tried to run the test python code, nothing was printed on the screen, leading me to find out that the form of `void util::DictAttributeSerializer::on_adapter(const std::string& name, ov::ValueAccessor<void>& adapter)` was actually invoked. Thats why I choose the `if-else` inside this specific function as the solution.

*And then I did some more tests.* 

I added a new `on_adapter()` function inside the base `AttributeVisitor` class:

```cpp
// src/core/include/openvino/core/attribute_visitor.hpp
class OPENVINO_API AttributeVisitor {
public:
    ...
    // This new function is not a virtual one to avoid implementing it among all its derived classes
    void on_adapter(const std::string& name, ValueAccessor<ov::PartialShape>& adapter) {std::cout << "on_adapter(base)" << std::endl;}
    ...
};
```

It turned out the function of the base class can be invoked instead of that of `DictAttributeSerializer` (its derived class). And im also not sure about whether the invasive modifications of files under `src/core` is allowed just for extending a python API.

Your suggestions would be really appreciated!